### DISCUSSION TITLE
aircrack-ng on NetBSD

### DISCUSSION THREAD

**Discussion by 0323pin** [aircrack-ng/aircrack-ng#2598] [Q&A] [answered]
Hi,

While trying to package [airgorah](https://github.com/martin-olivier/airgorah) for NetBSD, I needed to update the `aircrack-ng` package that is in our wip repository.

I must say, I don't know much about the tools provided by `aircrack-ng` but, while reading the documentation, I've realized that there's no `airmon-ng.netbsd` script in the [scripts directory](https://github.com/aircrack-ng/aircrack-ng/tree/master/scripts). This is a bit strange since, from what I've been reading this script is rather essential.

Any clues on how to use the tools without this script? Or, should we write our own?
Thank you.

**Comment by Mister-X-** [comment, marked as answer]
You have to put the adapter in monitor mode, then you can use the different tools.

I could write one, but I just don't have a netbsd machine handy with a wireless adapter to test.

**Reply by 0323pin** [reply]
> You have to put the adapter in monitor mode

Sorry but, I'm new to all this. How would I do that? Something like `iwconfig iface mode monitor` on Linux which, most probably translates to something like `wiconfig iface -D` on NetBSD?

**Comment by 0323pin** [comment]
@Mister-X- or, do you mean monitor mode specific to my wifi chip using the corresponding driver?

**Reply by Mister-X-** [reply]
I don't even know what adapter and driver you have, so no.

For Intel, with the [iwn](https://man.netbsd.org/iwn.4) driver, it's something like `ifconfig $INTERFACE mediaopts monitor`

**Reply by 0323pin** [reply]
Thanks!
It was just probably me being bad at explaining myself but, this is what I needed 👍

**Comment by 0323pin** [comment]
I should probably open an issue for this but, just in case it would have an easy answer ...

```
# ifconfig iwn0 mediaopt monitor
# airodump-ng iwn0
FAILED:src/airodump-ng/airodump-ng.c:627: pthread_mutex_lock(&(lopt.mx_print)) == 0
Exception: (core dumped)
~> gdb airodump-ng.core
GNU gdb (GDB) 13.2
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64--netbsd".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
airodump-ng.core: No such file or directory.
(gdb) bt
No stack.
```

**Reply by Mister-X-** [reply]
What version of aircrack-ng are you running? And what is the cpu architecture?

**Reply by 0323pin** [reply]
1.7 release built from source. I've updated the package myself on our wip-repository, https://github.com/NetBSD/pkgsrc-wip/tree/master/aircrack-ng

This is on NetBSD-current, amd64

**Reply by Mister-X-** [reply]
Can you give a try to latest version in git?

**Reply by 0323pin** [reply]
Yes, of course. I'll build later tonight and let you know.

**Comment by 0323pin** [comment]
Unfortunately, I get the same after building from 799d6f33bce7a6f61d08cf9db1acb1496743f0da

```
# ifconfig iwn0 mediaopt monitor
# airodump-ng iwn0
FAILED:src/airodump-ng/airodump-ng.c:687: pthread_mutex_lock(&(lopt.mx_print)) == 0
Abort (core dumped)
# gdb airodump-ng.core
GNU gdb (GDB) 13.2
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64--netbsd".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
"/home/pin/airodump-ng.core": not in executable format: file format not recognized
(gdb) bt
No stack.
```

**Reply by Mister-X-** [reply]
Could you open a bug? I'd have to investigate.

**Reply by 0323pin** [reply]
Will do, thanks!

**Comment by shadowkeke** [comment]
What is the state of this? Can be built on i386?

```
/usr/pkgsrc/aircrack-ng$ make
make: "/usr/pkgsrc/aircrack-ng/Makefile" line 21: Could not find ../../security/openssl/buildlink3.mk
make: "/usr/pkgsrc/aircrack-ng/Makefile" line 22: Could not find ../../databases/sqlite3/buildlink3.mk
make: "/usr/pkgsrc/aircrack-ng/Makefile" line 23: Could not find ../../devel/pcre2/buildlink3.mk
make: "/usr/pkgsrc/aircrack-ng/Makefile" line 24: Could not find ../../devel/cmocka/buildlink3.mk
make: "/usr/pkgsrc/aircrack-ng/Makefile" line 25: Could not find ../../devel/zlib/buildlink3.mk
make: "/usr/pkgsrc/aircrack-ng/Makefile" line 26: Could not find ../../mk/bsd.pkg.mk
make: Fatal errors encountered -- cannot continue
make: stopped in /usr/pkgsrc/aircrack-ng
```

On:

 10.1 NetBSD 10.1 (GENERIC) #0: Mon Dec 16 13:08:11 UTC 2024  mkrepro@mkrepro.NetBSD.org:/usr/src/sys/arch/i386/compile/GENERIC i386

Man that's good that what we have in Linux since about 15 years now just getting into netbsd :100:
